src/github.ts

```diff
@@ -547,6 +582,59 @@ export class GitHub {
     return tags;
   }
 
+  async findMergedPullRequests(number?: number): Promise<MergedGitHubPR[]> {
+    const targetBranch = await this.getDefaultBranch();
+    const response = await this.graphqlRequest({
+      query: `query findMergedPullRequests($owner: String!, $repo: String!, $num: Int!) {
```

Comment:
I was using this only for detecting the release PR/commit after the merge was completed and we're looking for the commit on the branch to tag (which in theory should be very close to head of the branch).

@bcoe pointed out that I missed the logic for fetching the commits since the last release on a target branch would also need to check the new branch names (which this PR is not yet doing).

For some reason, I thought that the REST endpoint couldn't filter by base branch, but [it totally can](https://docs.github.com/en/rest/reference/pulls#list-pull-requests). I will switch back to the REST endpoint

Reply:
Ah, I remember additional context - it fixes the ordering. Currently, we're sorting pull requests by `created` which is when the PR was opened. If a release PR has sat open for a long time with many PRs open, it may be buried in the list of closed pull requests and could require much paging to find. Coming from the commit side would give you the history order or when it's merged.

It might be possible to switch to `updated` for the sort order, but still a PR being modified after merge might affect this field too.

Reply:
> detecting the release PR/commit after the merge was completed and we're looking for the commit on the branch to tag

I actually think there should be one and only one way of finding this - whether it's for creating a tag or for creating/updating the next Release PR. The question is the same regardless of why it's being asked: "what is the commit on targetReleaseBranch that represents the last release" and I think we need to stick with the definition of "it's the last-merged-release-pr's merge commit". I think there should be a single `github.ts` method for this that has these inputs:

- targetReleaseBranch (defaultBranch/LTS branch)
- pattern for releasePRBranch name (constructed from your BranchName stuff with a regex placeholder for version if it's a `AutoreleaseBranchName` branch)

and finds the last merged PR given the above combo.

Reply:
The bug Ben pointed out is exactly this. The 2 need to use the same logic.

Reply:
I remember an additional reason why I was using commits to the branch to find the pull requests. The `base` ref of the associated pull request may not have been the release branch we're looking for.

Consider 2.0.0, 2.1.0 are both released from `main`. We now create a `2.0.x` branch. To find the latest release on the `2.0.x` branch, we need to consider *any* release pull request in history (the `2.0.0` release PR was against `main`, not the `2.0.x` branch - the `base` ref is not important).

Reply:
@chingor13 I want to make sure I understand what you're saying here:

Were versions `"2.0.0"` and `"2.1.0"` releases both release-please PRs that were merged to `main`? How does branch `2.0.x` come into existence? Is it manually created by branching off `main`? Ah, I think I get it. The problem is "bootstrapping" the initial version on the release branch, right? I.e. when release-please is configured for the first time with `defaultBranch` set to `2.0.x` it won't find the last release/version? I think we can solve this with the manifest file (thoughts forthcoming...) but till then does the `release-as` footer suffice as a work-around for that first release on the LTS branch?